apps/scan/src/test/scala/org/lfdecentralizedtrust/splice/scan/store/ScanEventStoreTest.scala

```diff
@@ -0,0 +1,762 @@
+        updateId2 = tx2.getUpdateId
+        _ <- insertVerdict(ctx1.verdictStore, updateId2, recordTs2, migrationId = mig1)
+
+        // Query combined events at current migration = mig1
+        events <- ctx1.eventStore.getEventsReference(
+          None,
+          mig1,
+          org.lfdecentralizedtrust.splice.store.PageLimit.tryCreate(10),
+        )
+        events2 <- ctx1.eventStore.getEventsReference(
+          Some((mig0, recordTs1.minusSeconds(1))),
+          mig1,
+          org.lfdecentralizedtrust.splice.store.PageLimit.tryCreate(10),
+        )
+        // after recordTs1
+        events3 <- ctx1.eventStore.getEventsReference(
+          Some((mig0, recordTs1)),
+          mig1,
+          org.lfdecentralizedtrust.splice.store.PageLimit.tryCreate(10),
+        )
+        // Fetch by id works across migrationIds
+        e1 <- ctx1.eventStore.getEventByUpdateId(updateId1)
+        e2 <- ctx1.eventStore.getEventByUpdateId(updateId2)
+      } yield {
+        // Cursor before ts1 should match no-cursor results
+        events shouldBe events2
+
+        // Should contain both updateIds
+        events.exists(_._2.exists(_.update.update.updateId == updateId1)) shouldBe true
+        events.exists(_._2.exists(_.update.update.updateId == updateId2)) shouldBe true
+
+        // Cursor at ts1 for mig0 should exclude mig0 event and include only mig1
+        events3.size shouldBe 1
+        events3.exists(_._2.exists(_.update.update.updateId == updateId1)) shouldBe false
+        events3.exists(_._2.exists(_.update.update.updateId == updateId2)) shouldBe true
+
+        // Fetch by id should yield both verdict and update
+        val (v1, u1) = e1.value
+        val (v2, u2) = e2.value
+
+        v1.value._1.migrationId shouldBe mig0
+        v2.value._1.migrationId shouldBe mig1
+
+        u1.value.migrationId shouldBe mig0
+        u2.value.migrationId shouldBe mig1
+      }
+    }
+
+    "provide assignment updates" in {
+      for {
+        ctx <- newEventStore()
+        recordTs = CantonTimestamp.now()
+
+        // Create an assignment (source -> target)
+        reassignment <- {
+          implicit val store: UpdateHistory = ctx.updateHistory
+          val sourceDomain = com.digitalasset.canton.topology.SynchronizerId
+            .tryFromString("source::domain")
+          val targetDomain = dummyDomain
+          targetDomain.assign(
+            (
+              appRewardCoupon(round = 0, provider = dsoParty, contractId = "assign-cid"),
+              sourceDomain,
+            ),
+            "rid-assign",
+            0,
+            recordTime = recordTs,
+          )
+        }
+
+        events <- ctx.eventStore.getEventsReference(
+          None,
+          domainMigrationId,
+          org.lfdecentralizedtrust.splice.store.PageLimit.tryCreate(10),
+        )
+
+        // currently does not provide reassignment updates, same behaviour as getUpdateByIdV2
```

Comment:
Note the `/v2/updates/{update_id}` endpoint does not provide reassignment updates. So at the moment we match its behavior